libraries/AP_Devo_Telem/AP_Devo_Telem.cpp

```diff
@@ -48,12 +48,12 @@ void AP_DEVO_Telem::init()
 }
 
 
-uint32_t AP_DEVO_Telem::gpsDdToDmsFormat(float ddm)
+uint32_t AP_DEVO_Telem::gpsDdToDmsFormat(int32_t ddm)
 {
-    int32_t deg = (int32_t)ddm;
-    float mm = (ddm - deg) * 60.0f;
+    int32_t deg = (int32_t)(ddm / 1e7);
+    double mm = (ddm / 1.0e7 - deg) * 60.0f;
 
-    mm = ((float)deg * 100.0f + mm) /100.0f;
+    mm = ((double)deg * 100.0f + mm) /100.0f;
```

Comment:
```suggestion
    mm = ((float)deg * 100.0f + mm) /100.0f;
```

The embedded hardware we run on doesn't necessarily have the hardware to run double-precision maths.  That puts a huge burden on the CPU as it has to emulate it in software.

Reply:
Done!